### PR TITLE
Docs: What’s new & Upgrade guide 10.4

### PR DESCRIPTION
**Pull request by imatwawana**
**PLEASE DO NOT ADD DIRECTLY TO THIS PR; CREATE A BRANCH AND MAKE CHANGES THERE FOR REVIEW PRIOR TO MERGING**

This PR adds the What’s new, which describes new features and changes available in Grafana v10.4, and the v10.4 Upgrade guide.

**Note for reviewers**: Review should be limited to placement of entries. Please do not make any copy edits. These entries are exactly the same as what's in the What's new in Cloud and making changes here creates divergence. If there's anything that looks wrong or is missing, please flag it that so we can decide on a course of action. 

**TO DO**

- [x] Add enablement videos as they are available - videos added but commented out. Will be commented in on a separate PR when videos are out of draft.

### CODE REVIEW

File: docs/sources/upgrade-guide/upgrade-v10.4/index.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+{{< docs/shared lookup="upgrade/upgrade-common-tasks.md" source="grafana" version="<GRAFANA VERSION>" >}}
+
+## Technical notes
+
+### Legacy alerting -> Grafana Alerting dry-run on start
+
+If you haven't already upgraded to Grafana Alerting from legacy Alerting, Grafana will initiate a dry-run of the upgrade every time the instance starts. This is in preparation for the removal of legacy Alerting in Grafana v11. The dry-run logs the results of the upgrade attempt and identifies any issues requiring attention before you can successfully execute the upgrade. No changes are made during the dry-run.
+
+You can disable this behaviour using the feature flag `alertingUpgradeDryrunOnStart`:
```
Comment (imatwawana) [inline docs/sources/upgrade-guide/upgrade-v10.4/index.md:28]: American English
```suggestion
You can disable this behavior using the feature flag `alertingUpgradeDryrunOnStart`:
```

**Comment by imatwawana** [comment]
@mjseaman - Note for review: just a reminder that we're not doing any copy editing; these entries (apart from any egregious typos) are exactly the same as what's in the What's new in Cloud and we don't want to create divergence. If there's anything that looks wrong, is missing, etc, then please flag that.

**Comment by mjseaman** [comment]
Thanks Isabel. 

I've added a longer intro paragraph and reordered some items for emphasis. I made a change before seeing your note that I should cut a separate PR - sorry! 

Some what's new posts use "you" and others use "users" to address the subject. IMO we should standardize on "you" per the [style guide](https://grafana.com/docs/writers-toolkit/write/style-guide/style-conventions/#address-users-clearly). I can go through the what's new posts and edit them there - is that the best course of action?

Also, I think we should include:
- the new Pagerduty data source - even though it's not technically part of the Grafana codebase, it's part of Grafana and this is a good place to announce it. https://admin.grafana.com/whats-new/#pagerduty-enterprise-data-source-for-grafana
- Return to previous. This one was marked for 10.4 - I wonder why it didn't make it into this PR.. Maybe it was submitted late? https://admin.grafana.com/content-admin/#/collections/whats-new/entries/2024-02-21-the-return-to-previous-functionality

I've also poked a couple of stragglers here to make sure significant updates are all in: https://raintank-corp.slack.com/archives/C036J5B39/p1709120845814739?thread_ts=1707928966.392599&cid=C036J5B39

**Comment by imatwawana** [comment]
> Thanks Isabel.
> 
> I've added a longer intro paragraph and reordered some items for emphasis. I made a change before seeing your note that I should cut a separate PR - sorry!

Making changes to the order and intro directly on the PR is fine. I just don't want folks adding entries to the PR directly.
> 
> Some what's new posts use "you" and others use "users" to address the subject. IMO we should standardize on "you" per the [style guide](https://grafana.com/docs/writers-toolkit/write/style-guide/style-conventions/#address-users-clearly). I can go through the what's new posts in the CMS and edit them there - is that the best course of action?

If you want to make copy edits to the posts, then yes, make those updates in the CMS and I'll implement those updates here when we generate the file for the last time tomorrow or Monday. Yes we should be using "you" vs "users", but that stuff should all be reviewed before folks merge their CMS entries. That's the behaviour we want to encourage: getting review (say during docs office hours, for example) before those entries ever make it to this side of things vs expecting cleanup at this stage where we've decided the docs team isn't going to do that anymore to keep this scalable.
> 
> Also, I think we should include:
> 
> * the new Pagerduty data source - even though it's not technically part of the Grafana codebase, it's part of Grafana and this is a good place to announce it. https://admin.grafana.com/whats-new/#pagerduty-enterprise-data-source-for-grafana

Okay, I'll add that in.

> * Return to previous. This one was marked for 10.4 - I wonder why it didn't make it into this PR.. Maybe it was submitted late? https://admin.grafana.com/content-admin/#/collections/whats-new/entries/2024-02-21-the-return-to-previous-functionality

Yes, that's likely what happened. But we'll generate another file from the What's new in Cloud tomorrow and probably again Monday to catch stragglers. I'll be merging this PR EOD Monday. Anything that comes in later than that will get added maybe about a week after that when I catch the last of the stragglers.
> 
> I've also poked a couple of stragglers here to make sure significant updates are all in: https://raintank-corp.slack.com/archives/C036J5B39/p1709120845814739?thread_ts=1707928966.392599&cid=C036J5B39

**Review by Eve832 — APPROVED** [review]

File: docs/sources/whatsnew/whats-new-in-v10-4.md [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+  - cloud
+  - enterprise
+  - oss
+title: What's new in Grafana v10.4
+weight: -41
+---
+
+# What’s new in Grafana v10.4
+
+Welcome to Grafana 10.4! This minor release contains some notable improvements in its own right, as well as early previews of functionality we intend to turn on by default in Grafana v11. Read on to learn about a quicker way to set up alert notifications, an all-new UI for configuring up single sign-on, and improvements to our Canvas, Geomap, and Table panels.
```
Comment (mjseaman) [inline docs/sources/whatsnew/whats-new-in-v10-4.md:20]:
```suggestion
Welcome to Grafana 10.4! This minor release contains some notable improvements in its own right, as well as early previews of functionality we intend to turn on by default in Grafana v11. Read on to learn about a quicker way to set up alert notifications, an all-new UI for configuring single sign-on, and improvements to our Canvas, Geomap, and Table panels.
```

File: docs/sources/whatsnew/whats-new-in-v10-4.md [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+
+We are working on adding complete support for configuring all other supported OAuth providers as well, such as GitHub, GitLab, Google, Microsoft Azure AD and Okta. You can already manage some of these settings via the new self-serve configuration options, and we’re working on adding more at the moment.
+
+![Screenshot of the Authentication provider list page](/media/docs/grafana-cloud/screenshot-sso-settings-ui-public-prev-v10.4.png)
+
+[Documentation](https://grafana.com/docs/grafana/next/setup-grafana/configure-security/configure-authentication/)
+
+## Data sources
+
+> Note: the following data sources are released separately from Grafana itself. They are included here for extra visibility.
```
Comment (imatwawana) [inline docs/sources/whatsnew/whats-new-in-v10-4.md:220]:
```suggestion
{{< admonition type="note" >}}
The following data sources are released separately from Grafana itself. They are included here for extra visibility.
{{< /admonition >}}
```

**Pull request merged**